zerver/decorator.py

```diff
@@ -159,7 +159,10 @@ def process_client(request: HttpRequest, user_profile: UserProfile,
                    skip_update_user_activity: bool=False,
                    query: Optional[str]=None) -> None:
     if client_name is None:
-        client_name = get_client_name(request, is_browser_view)
+        if hasattr(request, "client_name"):
+            client_name = request.client_name
+        else:
+            client_name = get_client_name(request, is_browser_view=is_browser_view)
```

Comment:
When do you expect this `else` case to happen?

Reply:
I am not sure why this happens but if I don't put the else part then during the tests I get:
```
Running zerver.tests.test_decorators.DecoratorLoggingTestCase.test_authenticated_rest_api_view_with_non_webhook_view

======================================================================
FAIL: test_authenticated_rest_api_view_with_non_webhook_view (zerver.tests.test_decorators.DecoratorLoggingTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.5/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/usr/lib/python3.5/unittest/case.py", line 600, in run
    testMethod()
  File "/home/circleci/zulip/zerver/tests/test_decorators.py", line 575, in test_authenticated_rest_api_view_with_non_webhook_view
    non_webhook_view_raises_exception(request)
  File "/usr/lib/python3.5/unittest/case.py", line 212, in __exit__
    expected_regex.pattern, str(exc_value)))
  File "/usr/lib/python3.5/unittest/case.py", line 134, in _raiseFailure
    raise self.test_case.failureException(msg)
AssertionError: "raised by a non-webhook view" does not match "'HostRequestMock' object has no attribute 'client_name'"

```
Most probably this test directly passes a request variable to process_client without calling the middleware. Will look into it.